src/app/profile/yourGiving/historicalView/historicalView.component.js

```diff
@@ -34,11 +40,56 @@ class HistoricalView {
       this.loadingGiftsError = true
     })
   }
+
+  isRecent (year, month) {
+    const today = moment()
+    return today.subtract(3, 'months').isBefore(moment(`${year}/${month}/01`, 'YYYY/MM/DD'))
+  }
+
+  parseHistoricalGifts (historicalGifts, year, month) {
+    if (!historicalGifts.length || !historicalGifts[0].donations) {
+      return historicalGifts
+    }
+
+    const filteredList = historicalGifts.filter((historicalGift) => {
+      historicalGift.donations = historicalGift.donations.filter((donation) => {
+        const transactionDate = donation['historical-donation-line']['transaction-date']
+        const momentDate = moment(transactionDate['display-value'], 'YYYY/MM/DD')
+
+        return momentDate.year() === year && (momentDate.month() + 1) === month
+      })
+      return historicalGift.donations.length > 0
+    })
+
+    return filteredList.flatMap((donationSummary) => {
+      if (donationSummary.donations) {
```

Comment:
This condition is always true right? If `donations` isn't set, I believe we'll get an exception on line 55.

Reply:
Yeah, I think so. Probably this is left over from my TDD work as I had both in place in parallel.